build.gradle

```diff
@@ -24,7 +24,15 @@ dependencies {
     implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.13.3'
 
     testImplementation gradleTestKit()
-    testImplementation 'org.spockframework:spock-core:2.1-groovy-3.0'
+
+    testImplementation(platform('org.apache.groovy:groovy-bom:4.0.4'))
+    testImplementation('org.spockframework:spock-core:2.2-groovy-4.0') {
+        exclude(group: 'org.codehaus.groovy')
+    }
+}
+
+tasks.named('compileTestGroovy').configure {
+    options.forkOptions.jvmArgs << '-Dspock.iKnowWhatImDoing.disableGroovyVersionCheck=true'
 }
```

Comment:
Could you confirm @leonard84 that this is the way to get `spock-groovy-4` used with _Gradle_ 6 or 7?

Reply:
You don't need that switch if you use aligned Spock/Groovy versions.

Reply:
Without the switch I'm getting a compilation error 
`IncompatibleGroovyVersionException: The Spock compiler plugin cannot execute because Spock 2.2.0-groovy-4.0 is not compatible with Groovy 3.0.10`
as Gradle brings in Groovy 3

See https://ge.solutions-team.gradle.com/s/ynzhqj7leuxhe/console-log?page=1#L30